tests/test_raytracing.py

```diff
@@ -0,0 +1,319 @@
+    bf_header = self.fp["Header"].attrs
+
+    ###############################################
+    #Translate FlowPM header into lenstools header#
+    ###############################################
+
+    #Number of particles/files
+    header["num_particles_file"] = bf_header["NC"][0]**3
+    header["num_particles_total"] = header["num_particles_file"]
+    header["num_files"] = 1
+
+    #Cosmology
+    header["Om0"] = bf_header["OmegaCDM"][0] + bf_header["OmegaB"][0]
+    header["Ode0"] = 1. - header["Om0"]
+    header["w0"] = -1.
+    header["wa"] = 0.
+    header["h"] = bf_header["h"][0]
+    header["redshift"] = 1 / (bf_header["Time"][0]) - 1
+    header["comoving_distance"] = bf_header["comoving_distance"][0] * 1.0e3
+    header["scale_factor"] = bf_header["Time"][0]
+
+    #Box size in kpc/h
+    header["box_size"] = bf_header["BoxSize"][0] * 1.0e3
+    header["box_size_mpch"] = bf_header["BoxSize"][0]
+    #Plane Resolution
+    header["nc"] = bf_header["NC"][0]
+    #Masses
+    header["masses"] = np.array(
+        [0., bf_header["M0"][0] * header["h"], 0., 0., 0., 0.])
+    #################
+
+    return header
+
+  def setLimits(self):
+
+    if self.pool is None:
+      self._first = None
+      self._last = None
+    else:
+
+      #Divide equally between tasks
+      Nt, Np = self.pool.size + 1, bigfile.BigData(self.fp).size
+      part_per_task = Np // Nt
+      self._first = part_per_task * self.pool.rank
+      self._last = part_per_task * (self.pool.rank + 1)
+
+      #Add the remainder to the last task
+      if (Np % Nt) and (self.pool.rank == Nt - 1):
+        self._last += Np % Nt
+
+  def getPositions(self, first=None, last=None, save=True):
+
+    #Get data pointer
+    data = self.fp
+
+    #Read in positions in Mpc/h
+    if (first is None) or (last is None):
+      positions = (
+          data["0/Position"][:] + np.array([
+              0.5 / self.header["nc"] * self.header["box_size_mpch"],
+              0.5 / self.header["nc"] * self.header["box_size_mpch"], 0
+          ],
+                                           dtype=np.float32)) * self.Mpc_over_h
+    else:
+      positions = data["0/Position"][first:last] * self.Mpc_over_h
+
+    #Enforce periodic boundary conditions
+    for n in (0, 1):
+      positions[:, n][positions[:, n] < 0] += self.header["box_size"]
+      positions[:, n][
+          positions[:, n] > self.header["box_size"]] -= self.header["box_size"]
+
+    #Maybe save
+    if save:
+      self.positions = positions
+
+    #Initialize useless attributes to None
+    self.weights = None
+    self.virial_radius = None
+    self.concentration = None
+
+    #Return
+    return positions
+
+
+def test_density_plane(return_results=False):
+  """ Tests cutting density planes from snapshots against lenstools
+  """
+  klin = np.loadtxt(data_path + '/flowpm/data/Planck15_a1p00.txt').T[0]
+  plin = np.loadtxt(data_path + '/flowpm/data/Planck15_a1p00.txt').T[1]
+  ipklin = iuspline(klin, plin)
```

Comment:
ah yes, it's the same, except this might be a little faster than using the `linear_matter_power`function. It's just to speed up the test a tiny bit. But you are perfectly right, we could have using tfpower ;-)